util/sem/sem.go

```diff
@@ -155,7 +155,9 @@ func IsInvisibleSysVar(varNameInLower string) bool {
 		variable.TiDBMemoryUsageAlarmRatio,
 		variable.TiDBRedactLog,
 		variable.TiDBRestrictedReadOnly,
-		variable.TiDBSlowLogMasking:
+		variable.TiDBSlowLogMasking,
```

Comment:
Is it an idiom that for variables that cannot be changed without RESTRICTED_VARIABLES_ADMIN they should be hidden? @morgo PTAL

Reply:
Currently yes. There are two problems:

* Some of these variables can reveal metadata (filesystem paths, hostnames etc).
* In the original design the PM wanted reading the variable to be an error, but there are no cases where MySQL would error on reading a variable, so that behavior makes compatibility weird.

So that is why the behavior of SEM is to hide the variables, even though for the server we are discontinuing the use of "hidden" variables.